ucp/useraccounts_module.php

```diff
@@ -308,154 +265,162 @@ function main($id, $mode)
 			case 'select_badges':
 			{
 				// Set up vars for mode.
-				$markedbadges = $this->request->variable('marked_badges', array(0)); // badge_ids
-				$submitmarked = $this->request->is_set_post('marked_badges') || empty($markedbadges) && $action == 'submit_marked' ? true : false;
+				$markedbadges = $this->request->variable('markedbadges', array(0)); // Badge IDs, automatically casted to an array of integers.
 
-				if ($submitmarked)
+				if ($this->request->is_set_post('markedbadges') || empty($markedbadges) && $submit )
```

Comment:
Need parens here too.  Mixing && and || in a statement without them is really bad practice.